packages/design-system/src/hooks/web3Wagmi.tsx

```diff
@@ -50,3 +53,28 @@ export function useEthersSigner({ chainId }: { chainId?: number } = {}) {
     [walletClient],
   )
 }
+
+export function publicClientToProvider(publicClient: PublicClient) {
+  const { chain, transport } = publicClient
+  const network = {
+    chainId: chain.id,
+    name: chain.name,
+    ensAddress: chain.contracts?.ensRegistry?.address,
+  }
+  if (transport.type === 'fallback')
```

Comment:
What does this condition check?  `if (transport.type === 'fallback')`

Reply:
that's a conditional to check if the public client was created using the fallback method from `viem`. If that's the case then we need to iterate over multiple public clients and also create an ethers fallback provider that also accepts multiple Providers. If not it means we just have one provider as a result we just need one ethers Provider.

You can refer to the docs for each library here:
- [Ethers FallbackProvider](https://docs.ethers.org/v5/api/providers/other/#:~:text=The%20FallbackProvider%20is%20the%20most,a%20priority%20and%20a%20weight%20.)
- [Viem transport type 'fallback'](https://viem.sh/docs/clients/transports/fallback.html)

Reply:
Thank you for the explanation, I will check the docs for more more info